spring-integration-core/src/main/java/org/springframework/integration/support/converter/DefaultDatatypeChannelMessageConverter.java

```diff
@@ -58,7 +59,7 @@ public void setConversionService(ConversionService conversionService) {
 	}
 
 	@Override
-	public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
+	public void setBeanFactory(@Nullable BeanFactory beanFactory) throws BeansException {
```

Comment:
Same here: we need to rework the logic with an assumption that it is not null.

Reply:
well in this case, we could just drop the check in the following if statement